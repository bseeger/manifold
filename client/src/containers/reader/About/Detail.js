import React, { PureComponent, PropTypes } from 'react';
import { textsAPI, requests } from 'api';
import { entityStoreActions } from 'actions';
import { entityUtils } from 'utils';
import { connect } from 'react-redux';
import { About } from 'components/reader';

const { select, meta } = entityUtils;
const { request, flush } = entityStoreActions;

class AboutDetailContainer extends PureComponent {

  static displayName = "ReaderContainer.About.Detail";

  static fetchData(getState, dispatch, location, params) {
    const promises = [];
    const textCall = textsAPI.show(params.textId);
    const { promise: one } = dispatch(request(textCall, requests.rText));
    promises.push(one);
    return Promise.all(promises);
  }

  static mapStateToProps(state, ownProps) {
    const newState = {
      text: select(requests.rText, state.entityStore),
      textMeta: meta(requests.rText, state.entityStore),
    };
    return Object.assign({}, newState, ownProps);
  }

  static propTypes = {
    route: PropTypes.object,
    params: PropTypes.object,
    text: PropTypes.object,
    dispatch: PropTypes.func
  };

  constructor(props) {
    super(props);
  }

  componentWillUnmount() {
    this.props.dispatch(flush(requests.rText));
  }

  render() {
    if (!this.props.text) return null;

    return (
      <About.Overlay
        params={this.props.params}
        text={this.props.text}
      />
    );
  }
}

export default connect(
  AboutDetailContainer.mapStateToProps
)(AboutDetailContainer);
