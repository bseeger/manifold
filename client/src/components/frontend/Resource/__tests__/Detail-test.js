import React from 'react';
import renderer from 'react-test-renderer';
import Detail from '../Detail';
import build from 'test/fixtures/build';
import { Provider } from 'react-redux';

describe("Frontend.Resource.Detail component", () => {

  const store = build.store();

  const resource = build.entity.resource("1");
  const project = build.entity.project("2");

  it('renders correctly', () => {
    const component = renderer.create(
      <Provider store={store} >
        <Detail
          resource={resource}
          projectId={project.id}
          resourceUrl="resource/url"
          projectUrl="project/url"
        />
      </Provider>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

});
