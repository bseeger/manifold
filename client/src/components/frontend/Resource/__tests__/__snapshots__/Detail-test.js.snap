exports[`Frontend.Resource.Detail component renders correctly 1`] = `
<div>
  <section>
    <section
      className="resource-detail">
      <div
        className="container flush-top flush-bottom">
        <div
          className="resource-title">
          <div>
            <h1
              dangerouslySetInnerHTML={
                Object {
                  "__html": "Image",
                }
              } />
            <span
              className="resource-date">
              Resource added 
              <span>
                April, 2017
              </span>
            </span>
          </div>
        </div>
      </div>
      <section
        className="resource-hero-container">
        <div
          className="resource-hero">
          <div
            className="resource-slide-figure">
            <figure>
              <div
                className="figure-image"
                style={
                  Object {
                    "backgroundImage": "url(null)",
                  }
                } />
            </figure>
          </div>
        </div>
      </section>
      <div
        className="container flush-top">
        <aside>
          <a
            className="button-primary"
            onClick={[Function]}
            style={Object {}}
            target="_blank">
            Download 
            <i
              className="manicon manicon-arrow-down" />
          </a>
          <nav
            className="share-nav-primary">
            <span>
              Share
            </span>
            <ul>
              <li>
                <button
                  onClick={[Function]}
                  target="_blank">
                  <i
                    className="manicon manicon-twitter" />
                </button>
              </li>
            </ul>
          </nav>
          <section
            className="resource-meta">
            <figure
              className="resource-type">
              <i
                className="manicon manicon-resource-image" />
            </figure>
            <ul
              className="meta-list-secondary">
              <li>
                <span
                  className="meta-label">
                  Type
                </span>
                <span
                  className="meta-value">
                  Image
                </span>
              </li>
              <li>
                <span
                  className="meta-label">
                  Created On
                </span>
                <span
                  className="meta-value">
                  <span>
                    April 24, 2017
                  </span>
                </span>
              </li>
            </ul>
            <nav
              className="tag-list">
              <ul>
                <li>
                  <a
                    onClick={[Function]}
                    style={Object {}}>
                    dog
                  </a>
                </li>
                <li>
                  <a
                    onClick={[Function]}
                    style={Object {}}>
                    puppy
                  </a>
                </li>
                <li>
                  <a
                    onClick={[Function]}
                    style={Object {}}>
                    GOAT
                  </a>
                </li>
              </ul>
            </nav>
          </section>
        </aside>
        <div
          className="resource-content left">
          <div
            dangerouslySetInnerHTML={
              Object {
                "__html": "World\'s Greatest Dog",
              }
            } />
          <h3
            className="attribute-header">
            Full Description
          </h3>
          <div
            dangerouslySetInnerHTML={
              Object {
                "__html": "Black and white freckles",
              }
            } />
        </div>
        <div
          className="resource-meta-mobile">
          <section
            className="resource-meta">
            <figure
              className="resource-type">
              <i
                className="manicon manicon-resource-image" />
            </figure>
            <ul
              className="meta-list-primary">
              <li>
                <span
                  className="meta-label">
                  Type
                </span>
                <span
                  className="meta-value">
                  Image
                </span>
              </li>
              <li>
                <span
                  className="meta-label">
                  Created On
                </span>
                <span
                  className="meta-value">
                  <span>
                    April 24, 2017
                  </span>
                </span>
              </li>
            </ul>
            <nav
              className="tag-list">
              <ul>
                <li>
                  <a
                    onClick={[Function]}
                    style={Object {}}>
                    dog
                  </a>
                </li>
                <li>
                  <a
                    onClick={[Function]}
                    style={Object {}}>
                    puppy
                  </a>
                </li>
                <li>
                  <a
                    onClick={[Function]}
                    style={Object {}}>
                    GOAT
                  </a>
                </li>
              </ul>
            </nav>
          </section>
        </div>
      </div>
    </section>
  </section>
</div>
`;
